sc-app-deploy/sad_deploy/deploy_commands.py

```diff
@@ -0,0 +1,105 @@
+from logging import exception
+import sys
+import os
+import subprocess
+import logging
+from contextlib import redirect_stdout
+import sad_common
+from sad_common.docker_helper import DockerRegistry
+
+from sad_common.run_command import runCommand
+from sad_infra.application import Application
+from sad_infra.host import Host
+import sad_secrets.secret_helper
+
+team_target_postfix = "schul-cloud.dev"
+auto_target_postfix = "schul-cloud.org"
+team_host_name_prefix = "hotfix"
+auto_host_name = "test"
+docker_namespace = "schulcloud"
+sc_image_list = [
+    {'image_name': "schulcloud-server", 'application_name': "server"},
+    {'image_name': "schulcloud-client", 'application_name': "client"},
+    {'image_name': "schulcloud-nuxt-client", 'application_name': "nuxtclient"},
+    {'image_name': "schulcloud-calendar", 'application_name': "calendar" }
+]
+
+def deployImage(application: Application, host: Host, decryptedSshKeyFile: str):
+    '''
+    Deploys the application to the given host.
+    '''
+    logging.info("Deploying '%s'..." % application.getSwarmServicename(host))
+
+    remoteUser = "travis"
+
+    sshCommand=['ssh']
+
+    # Disable known hosts checking.
+    sshOptions=['-o', 'StrictHostKeyChecking=no', '-o', 'UserKnownHostsFile=/dev/null']
+
+    if decryptedSshKeyFile != None:
+        # Use provided ssh key
+        sshIdentity=['-i', decryptedSshKeyFile]
+    else:
+        # Try pageant
+        sshIdentity=[]
+
+    # remote = <build user>@<hostname>.schul-cloud.dev
+    sshRemote=['%s@%s' % (remoteUser, host.getFQDN())]
+
+    # The remote command parameters.
+    # image = <imagename>:<imagetag> = <repository name>:<tag>
+    # service name = <hostname>_<applicationname short>. See 'docker service ls'
+    sshRemoteCommandParameters=[application.getImage(), application.getSwarmServicename(host)]
+
+    command = sshCommand + sshOptions + sshIdentity + sshRemote + sshRemoteCommandParameters
+    logging.info("Running command: '%s'" % ' '.join(command))
+
+    # Run docker service update
+    # Example call: ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i travisssh travis@hotfix6.schul-cloud.dev schulcloud/schulcloud-server:develop_latest hotfix6_server
+    # Example execution: /usr/bin/docker service update --force --image schulcloud/schulcloud-server:develop_latest hotfix6_server
+    runCommand(command)
+    #runCommand(['ls'])
+    logging.info("Deployment '%s' complete." % application.getSwarmServicename(host))
+
+    # TODO: Inform RocketChat
+
+def deployImages(branch, args):
```

Comment:
Add comment that describes what the method is doing. Especially when pulling the arguments parsing logic forward deeper into the applicaton logic. 

Making the arguments parser part of the program logic is not a good idea. If you would like to adept the command calls, you will have to dive down into the program logic.

Reply:
Changed to new paramter parsing